great_expectations/data_context/util.py

```diff
@@ -165,7 +166,12 @@ def substitute_config_variable(template_str, config_variables_dict):
             else:
                 return template_str[:match.start()] + config_variable_value + template_str[match.end():]
 
-        raise InvalidConfigError("Unable to find match for config variable {:s}. See https://great-expectations.readthedocs.io/en/latest/reference/data_context_reference.html#managing-environment-and-secrets".format(match.group(1)))
+        raise MissingConfigVariableError(
+                    f"""\n\nUnable to find a match for config substitution variable: `{match.group(1)}`.
+Please add this missing variable to your `uncommitted/config_variables.yml` file or your environment variables.
```

Comment:
Potentially clarify that the name uncommitted/config_variables isn't magical:

```suggestion
Please add this missing variable to your environment variables or config variables file (by default `uncommitted/config_variables.yml`).
```

Reply:
I totally forgot that this is configurable! Thanks!